packages/cli-repl/test/e2e.spec.ts

```diff
@@ -156,6 +156,20 @@ describe('e2e', function() {
         shell.assertContainsOutput('233');
       });
     });
+    it('accepts a --tlsFIPSMode argument', async() => {
+      shell = TestShell.start({
+        args: [ '--nodb', '--tlsFIPSMode' ]
+      });
+      const result = await shell.waitForPromptOrExit();
+      // Whether this worked depends on the environment the test is running in.
```

Comment:
Would it help to test here that the expected result is one which is expected for the environment? It might be extra coupling with the environment, is there a way we can do it without reproducing all of that?

Reply:
Well how do we know which result is the expected one? :) We could try to check it by enabling FIPS in the test as well and seeing what happens, but that feels like it would just be redundant compared to the code itself. (We could also check that this doesn’t work on macOS or Windows, which are systems where it is _very_ unlikely that we’re running this on a FIPS-enabled Node.js version, but even there that’s not really guaranteed)

[MONGOSH-1222](https://jira.mongodb.org/projects/MONGOSH/issues/MONGOSH-1222) will have an actual e2e test for FIPS mode, including makign sure that in those environments, we won’t get an error with `--tlsFIPSMode`.